src/resolvers/project.ts

```diff
@@ -48,6 +58,113 @@ export const resolvers: Resolvers = {
         throw InternalServerError();
       }
     },
+
+    searchExternalProjects: async (_, { affiliationId , awardId, awardName, awardYear, piNames }, context: MyContext): Promise<ExternalProject[]> => {
+      const reference = 'external project search resolver';
+
+      // Normalises dates
+      const normaliseDate = (date: string | null): string => {
+        if (date === null || date === undefined) {
+          return null;
+        }
+
+        // Split into parts, convert to integers and only continue if 3 integer parts
+        const parts = date
+          .split("-")
+          .map((p) => parseInt(p.trim(), 10))
+          .filter((p) => Number.isInteger(p));
+        if (parts.length !== 3) {
+          return null;
+        }
+
+        // Convert to numbers
+        return `${parts[0]}-${String(parts[1]).padStart(2, "0")}-${String(parts[2]).padStart(2, "0")}`;
+      };
+
+      const normaliseORCID = (input: string | null) => {
+        if (input === null || input === undefined) {
+          return null;
+        }
+        return input.trim().replace(/^https?:\/\/orcid\.org\//, '');
+      }
+
+      // Parses a contributor into a standard format
+      const parseContributor = (
```

Comment:
Please move this function into the `src/services/commonStandardService.ts`. I've got logic there that convert from our MySQL DB records into the JSON common standard format. I'm going to want to eventually have have functions to go in the other direction so this will be a good start

Reply:
Yeah sure, will do.